docs/design/architecture.md

```diff
@@ -221,70 +754,84 @@ or `container`. Kata Containers will then use this annotation to decide if it ne
 respectively create a virtual machine or a container inside a virtual machine associated
 with a Kubernetes pod:
 
-```Go
-	containerType, err := ociSpec.ContainerType()
-	if err != nil {
-		return err
-	}
-
-	handleFactory(ctx, runtimeConfig)
-
-	disableOutput := noNeedForOutput(detach, ociSpec.Process.Terminal)
-
-	var process vc.Process
-	switch containerType {
-	case vc.PodSandbox:
-		process, err = createSandbox(ctx, ociSpec, runtimeConfig, containerID, bundlePath, console, disableOutput, systemdCgroup)
-		if err != nil {
-			return err
-		}
-	case vc.PodContainer:
-		process, err = createContainer(ctx, ociSpec, containerID, bundlePath, console, disableOutput)
-		if err != nil {
-			return err
-		}
-	}
-
-```
+| Annotation value | Kata VM created? | Kata container created? |
+|-|-|-|
+| `sandbox` | yes | yes (inside new VM) |
+| `container`| no | yes (in existing VM) |
 
 #### Mixing VM based and namespace based runtimes
 
 > **Note:** Since Kubernetes 1.12, the [`Kubernetes RuntimeClass`](https://kubernetes.io/docs/concepts/containers/runtime-class/)
 > has been supported and the user can specify runtime without the non-standardized annotations.
 
-With `RuntimeClass`, users can define Kata Containers as a `RuntimeClass` and then explicitly specify that a pod being created as a Kata Containers pod. For details, please refer to [How to use Kata Containers and Containerd](../../docs/how-to/containerd-kata.md).
+With `RuntimeClass`, users can define Kata Containers as a
+`RuntimeClass` and then explicitly specify that a pod must be created
+as a Kata Containers pod. For details, please refer to [How to use
+Kata Containers and Containerd](../../docs/how-to/containerd-kata.md).
 
+## Tracing
+
+The [tracing document](../tracing.md) provides details on the tracing
+architecture.
 
 # Appendices
 
 ## DAX
 
-Kata Containers utilizes the Linux kernel DAX [(Direct Access filesystem)](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/filesystems/dax.rst?h=v5.14)
-feature to efficiently map some host-side files into the guest VM space.
-In particular, Kata Containers uses the QEMU NVDIMM feature to provide a
-memory-mapped virtual device that can be used to DAX map the virtual machine's
-root filesystem into the guest memory address space.
+Kata Containers utilizes the Linux kernel DAX
+[(Direct Access filesystem)](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/filesystems/dax.rst?h=v5.14)
+feature to efficiently map the [guest image](#guest-image) in the
+[host environment](#environments) into the
+[guest VM environment](#environments) to become the VM's
+[rootfs](#root-filesystem).
+
+If the [configured](#configuration) [hypervisor](#hypervisor) is set
+to either QEMU or Cloud Hypervisor, DAX is used by utilising the
+features as shown in the table below:
+
+| Hypervisor | Feature used |
+|-|-|
+| Cloud Hypervisor | `dax` `FsConfig` configuration option | 
```

Comment:
The statement that CLH is utilizing the DAX feature for the rootfs is correct, but it is actually using pmem device (instead of virtio-fs device) for the rootfs image. 

Also, technically the DAX option for virtio-fs now is disabled from the configuration-clh.toml [1] due to some instability issues related to virtio-fs. @jcvenegas Please correct me or add more comments if needed.

[1] https://github.com/kata-containers/kata-containers/blob/main/src/runtime/Makefile#L169-L170

Reply:
Hi @likebreath - Thanks for reviewing! I'm not quite clear what needs to change for your first point. Are you referring to the table in the "Environments" section? If so do you mean that the container rootfs needs to show DAX for CH?

I think we need another table like the following, but I'm wondering if that should be added to https://github.com/kata-containers/kata-containers/blob/main/docs/hypervisors.md rather than this doc:

| Hypervisor | VM rootfs mount type | container rootfs mount type |
|-|-|-|
| ACRN | ... | ... |
| CH | DAX | DAX |
| FC | ... | ... | 
| QEMU | DAX | virtio-fs |

Reply:
/cc @jcvenegas.